src/DependencyEmitter/AnnotationDependencyEmitter.php

```diff
@@ -0,0 +1,174 @@
+                    $annotation,
+                    $docComment,
+                    $property->getLine()
+                ));
+            }
+        }
+
+        return $buffer;
+    }
+
+    private function parseAnnotation(NikicPhpParser $astParser, AstFileReferenceInterface $fileReference, $annotation, $docComment, $line)
+    {
+        preg_match_all('/\* @' . $annotation . ' (.*)\\s/imsU', $docComment, $matches);
+        $classNames = isset($matches[1]) ? $matches[1] : null;
+
+        if (empty($classNames)) {
+            return [];
+        }
+
+        $buffer = [];
+
+        foreach ($classNames as $className) {
+            // @TODO maybe we could skip here values which are obvious no class
+            // names like string, integer etc.
+
+            // resolve the class name based on the defined uses if needed
+            $className = $this->resolveClassName($astParser, $fileReference, $className);
+            if (empty($className)) {
+                continue;
+            }
+
+            // check whether this class exists
+            $ast = $astParser->getAstForClassname($className);
+            if ($ast instanceof Class_) {
+                $buffer[] = new EmittedDependency(
+                    $className,
+                    $line,
+                    $annotation
+                );
+            }
+        }
+
+        return $buffer;
+    }
+
+    private function resolveClassName(NikicPhpParser $astParser, AstFileReferenceInterface $fileReference, $className)
+    {
+        // we have an fqcn
+        if (isset($className[0]) && $className[0] == '\\') {
+            return substr($className, 1);
+        }
+
+        // resolve class name based on defined uses
+        if (strpos($className, '\\') !== false) {
+            $alias = strstr($className, '\\', true);
+        } else {
+            $alias = $className;
+        }
+
+        $namespace = null;
+        foreach ($astParser->getAstByFile($fileReference) as $namespaceNode) {
+            if (!$namespaceNode instanceof Namespace_ || !$namespaceNode->stmts) {
+                continue; // @codeCoverageIgnore
+            }
+
+            $namespace = $namespaceNode->name->toString();
+            
+            foreach ($namespaceNode->stmts as $useNodes) {
+                if (!$useNodes instanceof Use_) {
+                    continue; // @codeCoverageIgnore
+                }
+
+                foreach ($useNodes->uses as $useNode) {
+                    if ($useNode->alias == $alias) {
+                        return $useNode->name->toString();
+                    }
+                }
+            }
+        }
+
+        if (!empty($namespace)) {
+            return Name::concat($namespace, $className)->toString();
+        }
+
+        return null;
+    }
+
+    private static function getDocBlocks($ast)
+    {
+        $visitor = new DocBlockVisitor();
+        $traverser = new NodeTraverser();
+        $traverser->addVisitor($visitor);
```

Comment:
can you inline variables that are used just once?
`(new NodeTraverser())->traverse(new NodeTraverser())`